airflow/callbacks/callback_requests.py

```diff
@@ -64,6 +73,34 @@ def __init__(
         self.simple_task_instance = simple_task_instance
         self.is_failure_callback = is_failure_callback
 
+    def __eq__(self, other):
+        if isinstance(other, TaskCallbackRequest):
+            return (
+                self.simple_task_instance.__dict__ == other.simple_task_instance.__dict__
+                and self.full_filepath == other.full_filepath
+                and self.is_failure_callback == other.is_failure_callback
+                and self.msg == other.msg
+            )
+        return False
```

Comment:
This should not be needed if we implement `CallbackRequest`’s `__eq__` “correctly”:

```python
def __eq__(self, other):
    if not isinstance(other, self.__class__):
        return NotImplemented
    return self.__dict__ == other.__dict__
```

Reply:
Right,I had to add __eq__ also for SimpleTaskInstance